pennylane/circuit_drawer/draw_mpl.py

```diff
@@ -0,0 +1,283 @@
+images.  If you change the docstring examples, please update this file.
+"""
+
+# cant `import pennylane as qml` because of circular imports with circuit graph
+from pennylane import ops
+from pennylane.wires import Wires
+from .mpldrawer import MPLDrawer
+from .drawable_layers import drawable_layers
+from .utils import convert_wire_order
+
+############################ Special Gate Methods #########################
+# If an operation is drawn differently than the standard box/ ctrl+box style
+# create a private method here and add it to the ``special_cases`` dictionary
+# These methods should accept arguments in the order of ``drawer, layer, mapped_wires, op``
+
+# pylint: disable=unused-argument
+def _add_swap(drawer, layer, mapped_wires, op):
+    drawer.SWAP(layer, mapped_wires)
+
+
+# pylint: disable=unused-argument
+def _add_cswap(drawer, layer, mapped_wires, op):
+    drawer.ctrl(layer, wires=mapped_wires[0], wires_target=mapped_wires[1:])
+    drawer.SWAP(layer, wires=mapped_wires[1:])
+
+
+# pylint: disable=unused-argument
+def _add_cx(drawer, layer, mapped_wires, op):
+    drawer.CNOT(layer, mapped_wires)
+
+
+def _add_multicontrolledx(drawer, layer, mapped_wires, op):
+    # convert control values
+    control_values = [(i == "1") for i in op.control_values]
+    drawer.ctrl(layer, mapped_wires[:-1], mapped_wires[-1], control_values=control_values)
+    # pylint: disable=protected-access
+    drawer._target_x(layer, mapped_wires[-1])
+
+
+# pylint: disable=unused-argument
+def _add_cz(drawer, layer, mapped_wires, op):
+    drawer.ctrl(layer, mapped_wires)
+
+
+special_cases = {
+    ops.SWAP: _add_swap,
+    ops.CSWAP: _add_cswap,
+    ops.CNOT: _add_cx,
+    ops.Toffoli: _add_cx,
+    ops.MultiControlledX: _add_multicontrolledx,
+    ops.CZ: _add_cz,
+}
+"""Dictionary mapping special case classes to functions for drawing them."""
+
+
+def draw_mpl(tape, wire_order=None, show_all_wires=False, decimals=None, **kwargs):
+    """Produces a matplotlib graphic from a tape.
+
+    Args:
+        tape (QuantumTape): the operations and measurements to draw
+
+    Keyword Args:
+        wire_order (Sequence[Any]): the order (from top to bottom) to print the wires of the circuit
+        show_all_wires (bool): If True, all wires, including empty wires, are printed.
+        decimals (int): How many decimal points to include when formatting operation parameters.  Default `None` will
+             omit parameters from operation labels.
+        wire_options (dict): matplotlib formatting options for the wire lines
+        label_options (dict): matplotlib formatting options for the wire labels
+
+    Returns:
+        matplotlib.figure.Figure, matplotlib.axes._axes.Axes: The key elements for matplotlib's object oriented interface.
+
+    **Example:**
+
+    .. code-block:: python
+
+        with qml.tape.QuantumTape() as tape:
+            qml.templates.GroverOperator(wires=(0,1,2,3))
```

Comment:
Don't need the .templates
```suggestion
            qml.GroverOperator(wires=(0,1,2,3))
```